src/renate/benchmark/models/l2p.py

```diff
@@ -0,0 +1,288 @@
+                    prompt_id = temp_pid
+
+                    temp_idc = torch.zeros((self._M,), device=id_counts.device)
+                    temp_idc[: id_counts.shape[0]] = id_counts
+                    id_counts = temp_idc
+
+                    _, major_idx = torch.topk(id_counts, k=self._N)
+                    idx = prompt_id[major_idx].expand(x.shape[0], -1)  # B, top_k
+            loss_value = similarity_matrix[:, idx].sum() / (x.shape[0] * x.shape[0])
+        else:
+            idx = manual_prompt_indices  # should be of size B, top_k
+            loss_value = torch.tensor(0.0, device=x.device)
+
+        selected_prompts = self.prompt_pool[idx].flatten(1, 2)
+        return selected_prompts, loss_value
+
+
+class PromptedVisionTransformer(RenateBenchmarkingModule):
+    """
+    Implements the vision transformer with prompt pool described in
+    Wang, Zifeng, et al. "Learning to prompt for continual learning." Proceedings of the
+    IEEE/CVF Conference on Computer Vision and Pattern Recognition. 2022.
+
+    Args:
+        pretrained_model_name_or_path: A string that denotes which pretrained model from the HF hub
+            to use. If provided, it overrides other arguments about architecture.
+        image_size: Size of the input image.
+        patch_size: Size of the patches.
+        num_layers: Number of Encoder layers.
+        num_heads: Number of Attention heads.
+        hidden_dim: Size of the Encoder's hidden state.
+        mlp_dim: Size of the intermediate Multi-layer Perceptron in the Encoder.
+        dropout: Dropout probability.
+        attention_dropout: Dropout probability for the attention in the Multi-head Attention layer.
+        num_outputs: Size of the output.
+        prediction_strategy: Continual learning strategies may alter the prediction at train or test
+            time.
+        add_icarl_class_means: If ``True``, additional parameters used only by the
+            ``ICaRLModelUpdater`` are added. Only required when using that updater.
+        pool_size: Total size of the prompt pool.
+        pool_selection_size: Number of prompts to select from the pool.
+        prompt_size: Number of tokens each prompt is equivalent to.
+        prompt_key_dim: Dimensions of the prompt key used to compute similarity. It has to be same
+            to the dimensions of `x` in forward.
+        embedding_dim: Output dimension of the token/patch embedding layer
+        train_prompt_keys: Whether to train the prompt keys. Currently unused.
+        similarity_fn: Similarity function between input features and prompt keys.
+        per_batch_prompt: Flag to use the same prompts for all elements in the batch.
+        prompt_embedding_features: Image feature type used to compute the similarity to prompt keys.
+        patch_pooler: Features to feed the classifier.
+    """
+
+    def __init__(
+        self,
+        pretrained_model_name_or_path="google/vit-base-patch16-224-in21k",
```

Comment:
any objection to make `google/vit-base-patch16-224` the default?

Reply:
No